src/client/driver/command-executors/execute-action-command.js

```diff
@@ -6,30 +6,70 @@ import {
-var positionUtils           = testCafeCore.positionUtils;
-var waitFor                 = testCafeCore.waitFor;
-var ClickAutomation         = testCafeRunner.get('./automation/playback/click');
-var RClickAutomation        = testCafeRunner.get('./automation/playback/rclick');
-var DblClickAutomation      = testCafeRunner.get('./automation/playback/dblclick');
-var DragToOffsetAutomation  = testCafeRunner.get('./automation/playback/drag/to-offset');
-var DragToElementAutomation = testCafeRunner.get('./automation/playback/drag/to-element');
-var HoverAutomation         = testCafeRunner.get('./automation/playback/hover');
-var TypeAutomation          = testCafeRunner.get('./automation/playback/type');
-var ProgressPanel           = testCafeUI.ProgressPanel;
+var Promise                         = hammerhead.Promise;
+var nativeMethods                   = hammerhead.nativeMethods;
+var XhrBarrier                      = testCafeCore.XhrBarrier;
+var pageUnloadBarrier               = testCafeCore.pageUnloadBarrier;
+var positionUtils                   = testCafeCore.positionUtils;
+var domUtils                        = testCafeCore.domUtils;
+var waitFor                         = testCafeCore.waitFor;
+var contentEditable                 = testCafeCore.contentEditable;
+var ClickAutomation                 = testCafeRunner.get('./automation/playback/click');
+var RClickAutomation                = testCafeRunner.get('./automation/playback/rclick');
+var DblClickAutomation              = testCafeRunner.get('./automation/playback/dblclick');
+var DragToOffsetAutomation          = testCafeRunner.get('./automation/playback/drag/to-offset');
+var DragToElementAutomation         = testCafeRunner.get('./automation/playback/drag/to-element');
+var HoverAutomation                 = testCafeRunner.get('./automation/playback/hover');
+var TypeAutomation                  = testCafeRunner.get('./automation/playback/type');
+var SelectTextAutomation            = testCafeRunner.get('./automation/playback/select/select-text');
+var SelectEditableContentAutomation = testCafeRunner.get('./automation/playback/select/select-editable-content');
+var getSelectPositionArguments      = testCafeRunner.get('./automation/playback/select/get-select-position-arguments');
+var ProgressPanel                   = testCafeUI.ProgressPanel;
+
+
+const PROGRESS_PANEL_TEXT      = 'Waiting for the target element of the next action to appear';
+const CHECK_ELEMENT_DELAY      = 200;
+const START_NODE_ARGUMENT_NAME = 'start node';
+const END_NODE_ARGUMENT_NAME   = 'end node';
+
+
+function ensureElementEditable (element) {
+    if (!domUtils.isEditableElement(element))
+        throw new ActionElementNonEditableError();
+
+    return element;
+}
+
+function ensureTextAreaElement (element) {
+    if (!domUtils.isTextAreaElement(element))
+        throw new ActionElementNotTextAreaError();
+
+    return element;
+}
 
+function ensureContentEditableNode (element, isStartNode) {
```

Comment:
Why we need the `isStartNode` argument and use an additional condition? We can just use `argumentName` arg here